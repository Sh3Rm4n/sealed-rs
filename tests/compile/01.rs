use sealed::sealed;

#[sealed]
pub struct A;
#[sealed]
pub struct B {
    field_1: i32,
}
pub struct C;

#[sealed]
trait T {}

impl T for A {}
impl T for B {}
impl T for C {}

fn main() {
    return;
}
