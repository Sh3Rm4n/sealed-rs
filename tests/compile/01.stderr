error[E0277]: the trait bound `C: Sealed` is not satisfied
  --> $DIR/01.rs:16:6
   |
11 | #[sealed]
   | --------- required by this bound in `T`
12 | trait T {}
   |       - required by a bound in this
...
16 | impl T for C {}
   |      ^ the trait `Sealed` is not implemented for `C`
